build/patch-charts.sh

```diff
@@ -38,42 +38,47 @@ function patchTemplates() {
   sed_wrap -i -e '/labels:/ i\
   annotations:\
     "maistra.io/internal": "true"' \
-    -e 's/name: istiod-{{.*/name: istiod-internal-{{ .Values.revision | default "default" }}/' \
-      ${HELM_DIR}/base/templates/role.yaml \
-      ${HELM_DIR}/base/templates/rolebinding.yaml
-  mv ${HELM_DIR}/base/templates/role.yaml ${HELM_DIR}/istio-control/istio-discovery/templates/role.yaml
-  mv ${HELM_DIR}/base/templates/rolebinding.yaml ${HELM_DIR}/istio-control/istio-discovery/templates/rolebinding.yaml
+    -e 's/^  name: istiod{{/  name: istiod-internal{{/' \
+      ${HELM_DIR}/istio-control/istio-discovery/templates/role.yaml \
+      ${HELM_DIR}/istio-control/istio-discovery/templates/rolebinding.yaml
 
   # update role reference too
-  sed_wrap -i -e 's/istiod-{{.*/istiod-{{ .Values.revision | default "default" }}-{{ .Release.Namespace }}/' \
-      ${HELM_DIR}/base/templates/clusterrolebinding.yaml
+#  sed_wrap -i -e 's/istiod-{{.*/istiod-{{ .Values.revision | default "default" }}-{{ .Release.Namespace }}/' \
+#      ${HELM_DIR}/base/templates/clusterrolebinding.yaml
 
-  mv ${HELM_DIR}/base/templates/serviceaccount.yaml ${HELM_DIR}/istio-control/istio-discovery/templates/serviceaccount.yaml
-  mv ${HELM_DIR}/base/templates/clusterrolebinding.yaml ${HELM_DIR}/istio-control/istio-discovery/templates/clusterrolebinding.yaml
+#  mv ${HELM_DIR}/base/templates/serviceaccount.yaml ${HELM_DIR}/istio-control/istio-discovery/templates/serviceaccount.yaml
+#  mv ${HELM_DIR}/base/templates/clusterrolebinding.yaml ${HELM_DIR}/istio-control/istio-discovery/templates/clusterrolebinding.yaml
 
   # - remove istio-reader cluster role
   # - and again....
-  sed_wrap -i -e '/^---/,$ d' \
-      -e 's/name: istiod-{{.*$/name: istiod-{{ .Values.revision | default "default" }}-{{ .Release.Namespace }}/' \
-      ${HELM_DIR}/base/templates/clusterrole.yaml
-  mv ${HELM_DIR}/base/templates/clusterrole.yaml ${HELM_DIR}/istio-control/istio-discovery/templates/clusterrole.yaml
+#  sed_wrap -i -e '/^---/,$ d' \
+#      -e 's/name: istiod-{{.*$/name: istiod-{{ .Values.revision | default "default" }}-{{ .Release.Namespace }}/' \
+#      ${HELM_DIR}/istio-control/istio-discovery/templates/clusterrole.yaml
+#  mv ${HELM_DIR}/base/templates/clusterrole.yaml ${HELM_DIR}/istio-control/istio-discovery/templates/clusterrole.yaml
 
   # - move istiod specific templates to istio-discovery
-  mv ${HELM_DIR}/base/templates/endpoints.yaml ${HELM_DIR}/base/templates/services.yaml ${HELM_DIR}/base/templates/validatingwebhookconfiguration.yaml ${HELM_DIR}/istio-control/istio-discovery/templates/
+#  mv ${HELM_DIR}/base/templates/endpoints.yaml ${HELM_DIR}/base/templates/services.yaml ${HELM_DIR}/base/templates/validatingwebhookconfiguration.yaml ${HELM_DIR}/istio-control/istio-discovery/templates/
+
+  sed_wrap -i -e 's/"cluster.local"/{{ .Values.global.trustDomain | default "cluster.local" }}/g' ${HELM_DIR}/istio-control/istio-discovery/templates/configmap.yaml
```

Comment:
nit: a comment would be good to understand why we need this (I'm guessing we should rather write the trustDomain directly into the meshconfig field instead of going through values.global in the future?)

Reply:
Done.